backend/inspirehep/migrator/tasks.py

```diff
@@ -344,11 +343,16 @@ def migrate_record_from_mirror(
     logger = LOGGER.bind(recid=prod_record.recid)
     try:
         json_record = marcxml2record(prod_record.marcxml)
+    except NotImplementedError as exc:
```

Comment:
This works because the `NotSupportedError` exception introduced in https://github.com/inspirehep/inspire-dojson/pull/261 inherits from `NotImplementedError`, but it doesn't follow the spirit of the exception translation pattern that you probably had in mind.

In fact, I'd argue that exception translation is an antipattern in a language without checked exceptions like Python. In Java, the compiler will check for you that any user of a function returning type `Foo` and throwing checked exception `Bar` does either catch the exception or throw it again. In Python, exceptions are not part of the signature (even when using any of the available typing mechanisms!), so you don't gain additional safety, but you have to write more boilerplate.

Reply:
That sounds very interesting but in this case I don't see any issue in applying polymorphism principles, it's one of the basics of OOP. 
Indeed, it works because `NotSupportedError` inherits from `NotImplementedError`, that's exactly the sole purpose of that class.

Reply:
On the other hand, being explicit and specific is another basic principle of Python. I don't think catching `NotImplementedError` will ever catch errors that are not `NotSupportedError`, but the devil is in the details so I guess explicit is better than implicit.

Reply:
That's not the point I'm trying to make: all I'm saying is that the exception translation pattern works in Java because you have checked exceptions, but is less useful in Python because it doesn't have them.

Another example is https://github.com/inspirehep/inspirehep/blob/4d73b2073a9c2de64ef344e481d7357ba11a7c79/backend/inspirehep/records/views.py#L85-L103
If someone modifies `import_article` so that it raises a new `FooException` in certain cases, no type checker is going to complain about it. One possible solution would be to use a union type as the return value, or to just ignore the problem and hope for the best. I'm claiming that exception translation is an antipattern in Python because it achieves the worst of both worlds: you end up having to write a bunch of boilerplate types, but you don't get any additional type safety out of it.